python/pyspark/errors/tests/test_traceback.py

```diff
@@ -0,0 +1,248 @@
+        try:
+            df.show()
+        except PythonException:
+            _, _, tb = sys.exc_info()
+        else:
+            self.fail("PythonException not raised")
+
+        s = "".join(traceback.format_tb(tb))
+        self.assertIn("""df.show()""", s)
+        self.assertIn("""raise ValueError("bar")""", s)
+
+    def test_datasource_analysis(self):
+        class MyDataSource(DataSource):
+            def schema(self):
+                raise ValueError("bar")
+
+        self.spark.dataSource.register(MyDataSource)
+        try:
+            self.spark.read.format("MyDataSource").load().show()
+        except AnalysisException:
+            _, _, tb = sys.exc_info()
+        else:
+            self.fail("AnalysisException not raised")
+
+        s = "".join(traceback.format_tb(tb))
+        self.assertIn("""self.spark.read.format("MyDataSource").load().show()""", s)
+        self.assertIn("""raise ValueError("bar")""", s)
+
+    def test_datasource_execution(self):
+        class MyDataSource(DataSource):
+            def schema(self):
+                return "x int"
+
+            def reader(self, schema):
+                return MyDataSourceReader()
+
+        class MyDataSourceReader(DataSourceReader):
+            def read(self, partitions):
+                raise ValueError("bar")
+
+        self.spark.dataSource.register(MyDataSource)
+        try:
+            self.spark.read.format("MyDataSource").load().show()
+        except PythonException:
+            _, _, tb = sys.exc_info()
+        else:
+            self.fail("PythonException not raised")
+
+        s = "".join(traceback.format_tb(tb))
+        self.assertIn("""self.spark.read.format("MyDataSource").load().show()""", s)
+        self.assertIn("""raise ValueError("bar")""", s)
+
+    def test_udtf_analysis(self):
+        @F.udtf()
+        class MyUdtf:
+            @staticmethod
+            def analyze():
+                raise ValueError("bar")
+
+            def eval(self):
+                ...
+
+        try:
+            MyUdtf().show()
+        except AnalysisException:
+            _, _, tb = sys.exc_info()
+        else:
+            self.fail("AnalysisException not raised")
+
+        s = "".join(traceback.format_tb(tb))
+        self.assertIn("""MyUdtf().show()""", s)
+        self.assertIn("""raise ValueError("bar")""", s)
+
+    def test_udtf_execution(self):
+        @F.udtf(returnType="x int")
+        class MyUdtf:
+            def eval(self):
+                raise ValueError("bar")
+
+        try:
+            MyUdtf().show()
+        except PythonException:
+            _, _, tb = sys.exc_info()
+        else:
+            self.fail("PythonException not raised")
+
+        s = "".join(traceback.format_tb(tb))
+        self.assertIn("""MyUdtf().show()""", s)
+        self.assertIn("""raise ValueError("bar")""", s)
+
+
+class TracebackSqlClassicTests(BaseTracebackSqlTestsMixin, ReusedSQLTestCase):
+    ...
```

Comment:
Let's do `pass` over `...`. I think we use `pass` more often in our codebase in this case